rust/blockstore/src/arrow/block/types.rs

```diff
@@ -118,16 +119,67 @@ impl Block {
         delta
     }
 
+    /// Binary search the block to find the last index of the specified prefix.
```

Comment:
nit: helpful to describe what the method does more formally such as

This API returns the largest index i s.t. prefix_arr[i] == target_prefix and None if prefix does not exist